atlasdb-client/src/main/java/com/palantir/atlasdb/logging/SimpleKeyValueServiceArgSupplier.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.logging;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.table.description.NameComponentDescription;
+import com.palantir.atlasdb.table.description.NamedColumnDescription;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+
+public class SimpleKeyValueServiceArgSupplier implements KeyValueServiceArgSupplier {
```

Comment:
Yep. Now you mention this I'm not sure we need to have multiple implementations - I was thinking in the event you want to apply some kind of logic that changes how you return safe/unsafe (e.g. live schema changes?), but you can achieve that by updating the arbiter as opposed to the switching logic. I'll change this.